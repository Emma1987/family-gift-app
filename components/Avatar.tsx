import ReactNiceAvatar, { genConfig } from '@zamplyy/react-native-nice-avatar';
import { StyleSheet, TouchableOpacity } from 'react-native';
import { Ionicons } from '@expo/vector-icons';

import { ThemedText } from '@/components/ThemedText';
import { ThemedView } from '@/components/ThemedView';
import { useThemeColor } from '@/hooks/useThemeColor';
import { AvatarComponentProps } from '@/types/appTypes';

export function Avatar({ type, familyMember, onPressCreateLabel = '', onPressCreate }: AvatarComponentProps) {
    const buttonBackgroundColor = useThemeColor({}, 'buttonBackgroundColor');
    const buttonTextColor = useThemeColor({}, 'buttonTextColor');

    if (type === 'blank-space') {
        return <ThemedView style={{ width: 100, height: 100 }} testID="blank-space"></ThemedView>;
    }

    if (type === 'create' && onPressCreate) {
        return (
            <TouchableOpacity onPress={onPressCreate} accessibilityLabel={onPressCreateLabel ?? ''}>
                <ThemedView style={[styles.button, { backgroundColor: buttonBackgroundColor }]}>
                    <Ionicons name="add" size={50} color={buttonTextColor} />
                </ThemedView>
            </TouchableOpacity>
        );
    }

    if (type === 'avatar' && familyMember?.avatar) {
        let config = genConfig();
        try {
            config = JSON.parse(familyMember.avatar);
        } catch (error) {
            console.error('Invalid avatar config:', error);
        }

        return (
            <ThemedView style={styles.avatar}>
                <ReactNiceAvatar size={100} {...config} />
                <ThemedText style={styles.name}>{familyMember.name}</ThemedText>
            </ThemedView>
        );
    }

    return null;
}

const styles = StyleSheet.create({
    button: {
        width: 100,
        height: 100,
        borderRadius: 50,
        justifyContent: 'center',
        alignItems: 'center',
    },
    avatar: {
        alignItems: 'center',
    },
    name: {
        textAlign: 'center',
        flexWrap: 'wrap',
        maxWidth: 100,
    },
});
