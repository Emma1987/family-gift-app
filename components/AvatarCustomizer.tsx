import React, { useEffect, useState } from 'react';
import { StyleSheet, View } from 'react-native';
import ReactNiceAvatar, { AvatarFullConfig } from '@zamplyy/react-native-nice-avatar';

import { Button } from '@/components/ui/Button';
import { ThemedText } from './ThemedText';
import {
    genderChoices,
    earSizeChoices,
    hairStyleChoices,
    eyeBrowStyleChoices,
    eyeStyleChoices,
    glassesStyleChoices,
    hatStyleChoices,
    mouthStyleChoices,
    noseStyleChoices,
    shirtStyleChoices,
    faceColorChoices,
    hairColorChoices,
    shirtColorChoices,
    hatColorChoices,
    bgColorChoices,
} from '@/constants/Avatar';

type AvatarCustomizerProps = {
    onAvatarChange: (config: any) => void;
};

export function AvatarCustomizer({ onAvatarChange }: AvatarCustomizerProps) {
    const [avatarConfig, setAvatarConfig] = useState<AvatarFullConfig>({
        sex: genderChoices[0],
        earSize: earSizeChoices[0],
        hairStyle: hairStyleChoices[0],
        eyeBrowStyle: eyeBrowStyleChoices[0],
        eyeStyle: eyeStyleChoices[0],
        glassesStyle: glassesStyleChoices[0],
        hatStyle: hatStyleChoices[0],
        mouthStyle: mouthStyleChoices[0],
        noseStyle: noseStyleChoices[0],
        shirtStyle: shirtStyleChoices[0],
        faceColor: faceColorChoices[0],
        hairColor: hairColorChoices[0],
        shirtColor: shirtColorChoices[0],
        hatColor: hatColorChoices[0],
        bgColor: bgColorChoices[0],
    });

    const cycleAvatarConfigKey = <T extends keyof AvatarFullConfig>(key: T, choices: AvatarFullConfig[T][]) => {
        setAvatarConfig((prevConfig) => {
            const currentIndex = choices.indexOf(prevConfig[key]);
            const nextIndex = (currentIndex + 1) % choices.length;

            return { ...prevConfig, [key]: choices[nextIndex] };
        });
    };

    useEffect(() => {
        onAvatarChange(avatarConfig);
    }, [avatarConfig]);

    return (
        <View>
            {/* Avatar */}
            <View style={styles.avatarContainer}>
                <ReactNiceAvatar size={150} {...avatarConfig} />
            </View>

            {/* Face options */}
            <View style={styles.optionsContainer}>
                <ThemedText style={styles.optionsTitle}>Visage</ThemedText>
                <View style={styles.buttonsContainer}>
                    <Button
                        text={'💁🏼‍♀️'}
                        accessibilityLabel={'Changer la couleur de peau'}
                        size="small"
                        onButtonPress={() => cycleAvatarConfigKey('faceColor', faceColorChoices)}
                    />
                    <Button
                        text={'👂🏽'}
                        accessibilityLabel={'Changer la forme des oreilles'}
                        size="small"
                        onButtonPress={() => cycleAvatarConfigKey('earSize', earSizeChoices)}
                    />
                    <Button
                        text={'🧔🏽‍♀️'}
                        accessibilityLabel={'Changer le style de sourcils'}
                        size="small"
                        onButtonPress={() => cycleAvatarConfigKey('eyeBrowStyle', eyeBrowStyleChoices)}
                    />
                    <Button
                        text={'👁️'}
                        accessibilityLabel={'Changer la forme des yeux'}
                        size="small"
                        onButtonPress={() => cycleAvatarConfigKey('eyeStyle', eyeStyleChoices)}
                    />
                    <Button
                        text={'👓'}
                        accessibilityLabel={'Ajouter des lunettes'}
                        size="small"
                        onButtonPress={() => cycleAvatarConfigKey('glassesStyle', glassesStyleChoices)}
                    />
                    <Button
                        text={'👄'}
                        accessibilityLabel={'Changer la forme de la bouche'}
                        size="small"
                        onButtonPress={() => cycleAvatarConfigKey('mouthStyle', mouthStyleChoices)}
                    />
                    <Button
                        text={'👃🏽'}
                        accessibilityLabel={'Changer la forme du nez'}
                        size="small"
                        onButtonPress={() => cycleAvatarConfigKey('noseStyle', noseStyleChoices)}
                    />
                </View>
            </View>

            {/* Hair options */}
            <View style={styles.optionsContainer}>
                <ThemedText style={styles.optionsTitle}>Cheveux</ThemedText>
                <View style={styles.buttonsContainer}>
                    <Button
                        text={'💇🏽‍♀️'}
                        accessibilityLabel={'Changer le style de coiffure'}
                        size="small"
                        onButtonPress={() => cycleAvatarConfigKey('hairStyle', hairStyleChoices)}
                    />
                    <Button
                        text={'👩🏼‍🦰'}
                        accessibilityLabel={'Changer la couleur de cheveux'}
                        size="small"
                        onButtonPress={() => cycleAvatarConfigKey('hairColor', hairColorChoices)}
                    />
                </View>
            </View>

            {/* Tee-shirt options */}
            <View style={styles.optionsContainer}>
                <ThemedText style={styles.optionsTitle}>Tee shirt</ThemedText>
                <View style={styles.buttonsContainer}>
                    <Button
                        text={'👕'}
                        accessibilityLabel={'Changer le style de tee shirt'}
                        size="small"
                        onButtonPress={() => cycleAvatarConfigKey('shirtStyle', shirtStyleChoices)}
                    />
                    <Button
                        text={'🔵'}
                        accessibilityLabel={'Changer la couleur de tee shirt'}
                        size="small"
                        onButtonPress={() => cycleAvatarConfigKey('shirtColor', shirtColorChoices)}
                    />
                </View>
            </View>

            {/* Hat options */}
            <View style={styles.optionsContainer}>
                <ThemedText style={styles.optionsTitle}>Chapeau</ThemedText>
                <View style={styles.buttonsContainer}>
                    <Button
                        text={'🎩'}
                        accessibilityLabel={'Changer le style de chapeau'}
                        size="small"
                        onButtonPress={() => cycleAvatarConfigKey('hatStyle', hatStyleChoices)}
                    />
                    <Button
                        text={'🔴'}
                        accessibilityLabel={'Changer la couleur de chapeau'}
                        size="small"
                        onButtonPress={() => cycleAvatarConfigKey('hatColor', hatColorChoices)}
                    />
                </View>
            </View>

            {/* Background options */}
            <View style={styles.optionsContainer}>
                <ThemedText style={styles.optionsTitle}>Fond</ThemedText>
                <View style={styles.buttonsContainer}>
                    <Button
                        text={'🟠'}
                        accessibilityLabel={'Changer la couleur de fond'}
                        size="small"
                        onButtonPress={() => cycleAvatarConfigKey('bgColor', bgColorChoices)}
                    />
                </View>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    avatarContainer: {
        alignItems: 'center',
        marginBottom: 20,
    },
    optionsContainer: {
        flexDirection: 'row',
    },
    optionsTitle: {
        width: '30%',
    },
    buttonsContainer: {
        width: '70%',
        flexDirection: 'row',
        flexWrap: 'wrap',
        gap: 5,
        marginBottom: 5,
        alignItems: 'center',
    },
});
