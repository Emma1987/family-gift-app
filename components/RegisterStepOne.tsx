import { StyleSheet, View } from 'react-native';
import { useFormContext } from 'react-hook-form';

import { Button } from '@/components/ui/Button';
import { PasswordInput } from '@/components/ui/PasswordInput';
import { TextInput } from '@/components/ui/TextInput';
import { ThemedText } from '@/components/ThemedText';

type RegisterStepOneProps = {
    nextStep: () => void;
};

export function RegisterStepOne({ nextStep }: RegisterStepOneProps) {
    const { control, trigger } = useFormContext();

    const handleNextStep = async () => {
        const validationResult = await trigger();
        if (validationResult) {
            nextStep();
        }
    };

    return (
        <View style={styles.container}>
            <ThemedText type="subtitle" style={styles.title}>
                Étape 1 : Informations
            </ThemedText>

            <TextInput
                name="name"
                control={control}
                placeholder="Nom"
                accessibilityLabel="Nom"
                rules={{
                    required: 'Le nom est obligatoire',
                    maxLength: { value: 150, message: 'Le nom ne doit pas dépasser 150 caractères' },
                }}
            />

            <TextInput
                name="email"
                control={control}
                placeholder="Adresse email"
                accessibilityLabel="Adresse email"
                rules={{
                    required: "L'email est obligatoire.",
                    pattern: { value: /^[^@]+@[^@]+\.[^@]+$/, message: "Cet email n'est pas un email valide" },
                    maxLength: { value: 180, message: "L'adresse email ne doit pas dépasser 180 caractères" },
                }}
                keyboardType="email-address"
            />

            <PasswordInput
                name="password"
                control={control}
                placeholder="Mot de passe"
                accessibilityLabel="Mot de passe"
                rules={{
                    required: 'Le mot de passe est obligatoire.',
                    pattern: {
                        value: /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[A-Za-z\d\W_]{8,}$/,
                        message:
                            "Le mot de passe doit être composé d'au minimum 8 caractères, et comporter au moins 1 lettre majuscule, 1 lettre minuscule et 1 chiffre.",
                    },
                }}
                autocomplete="password-new"
            />

            <View style={styles.buttonsContainer}>
                <Button text="Suivant" accessibilityLabel="Page suivante" onButtonPress={handleNextStep} />
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        marginTop: '25%',
    },
    title: {
        marginBottom: 20,
    },
    buttonsContainer: {
        alignItems: 'flex-end',
    },
});
