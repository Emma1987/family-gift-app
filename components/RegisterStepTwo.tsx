import React, { useMemo } from 'react';
import { StyleSheet, View } from 'react-native';
import { useFormContext } from 'react-hook-form';
import RadioGroup from 'react-native-radio-buttons-group';

import { Button } from '@/components/ui/Button';
import { TextInput } from '@/components/ui/TextInput';
import { ThemedText } from '@/components/ThemedText';
import { useThemeColor } from '@/hooks/useThemeColor';

enum FamilyChoiceEnum {
    JOIN = 'join-family',
    CREATE = 'create-family',
}

type RegisterStepTwoProps = {
    nextStep: () => void;
    prevStep: () => void;
};

export function RegisterStepTwo({ nextStep, prevStep }: RegisterStepTwoProps) {
    const inputStyle = useThemeColor({}, 'input');
    const { control, setValue, watch, trigger } = useFormContext();

    const selectedChoice = watch('selectedChoice', null);

    const familyChoices = useMemo(
        () => [
            {
                id: FamilyChoiceEnum.JOIN,
                label: 'Rejoindre ma famille grâce au code famille',
                value: FamilyChoiceEnum.JOIN,
                borderColor: inputStyle.borderColor,
                color: inputStyle.color,
                labelStyle: { color: inputStyle.color },
            },
            {
                id: FamilyChoiceEnum.CREATE,
                label: 'Créer une nouvelle famille',
                value: FamilyChoiceEnum.CREATE,
                borderColor: inputStyle.borderColor,
                color: inputStyle.color,
                labelStyle: { color: inputStyle.color },
            },
        ],
        [inputStyle],
    );

    const handleNextStep = async () => {
        const isStepValid = await trigger();
        if (isStepValid) {
            nextStep();
        }
    };

    return (
        <View style={styles.container}>
            <ThemedText type="subtitle" style={styles.title}>
                Étape 2 : Famille
            </ThemedText>

            <RadioGroup
                radioButtons={familyChoices}
                onPress={(choice) => setValue('selectedChoice', choice)}
                selectedId={selectedChoice}
                containerStyle={styles.radioContainer}
            />

            {selectedChoice === FamilyChoiceEnum.JOIN && (
                <TextInput
                    name="familyCode"
                    control={control}
                    placeholder="Code famille"
                    accessibilityLabel="Code de la famille"
                    rules={{
                        required: 'Le code famille est obligatoire',
                        pattern: {
                            value: /^[A-Z0-9]{6}$/,
                            message: "Ce code famille n'est pas un code famille valide",
                        },
                    }}
                />
            )}

            {selectedChoice === FamilyChoiceEnum.CREATE && (
                <TextInput
                    name="familyName"
                    control={control}
                    placeholder="Nom de la famille"
                    accessibilityLabel="Nom de la famille"
                    rules={{
                        required: 'Un nom pour créer la famille est obligatoire.',
                        maxLength: { value: 150, message: 'Le nom de la famille ne doit pas dépasser 150 caractères' },
                    }}
                />
            )}

            <View style={styles.buttonsContainer}>
                <Button text="Précédent" accessibilityLabel="Page précédente" onButtonPress={prevStep} />
                <Button text="Suivant" accessibilityLabel="Page suivante" onButtonPress={handleNextStep} />
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        marginTop: 100,
    },
    title: {
        marginBottom: 20,
    },
    radioContainer: {
        alignItems: 'flex-start',
        marginBottom: 20,
    },
    buttonsContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
});
