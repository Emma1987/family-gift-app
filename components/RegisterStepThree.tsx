import { StyleSheet, View } from 'react-native';
import { useFormContext } from 'react-hook-form';
import { AvatarFullConfig } from '@zamplyy/react-native-nice-avatar';

import { Button } from '@/components/ui/Button';
import { AvatarCustomizer } from '@/components/AvatarCustomizer';
import { ThemedText } from '@/components/ThemedText';

type RegisterStepThreeProps = {
    handleSubmit: () => void;
    prevStep: () => void;
};

export function RegisterStepThree({ handleSubmit, prevStep }: RegisterStepThreeProps) {
    const { setValue, watch, trigger } = useFormContext();

    const avatarConfig = watch('avatarConfig', null);

    const handleAvatarChange = (newConfig: AvatarFullConfig) => {
        setValue('avatarConfig', newConfig);
    };

    const handleNextStep = async () => {
        const isStepValid = await trigger();
        if (isStepValid) {
            handleSubmit();
        }
    };

    return (
        <View style={styles.container}>
            <ThemedText type="subtitle" style={styles.title}>
                Étape 3 : Avatar
            </ThemedText>

            <AvatarCustomizer onAvatarChange={handleAvatarChange} />

            <View style={styles.buttonsContainer}>
                <Button text="Précédent" accessibilityLabel="Page précédente" onButtonPress={prevStep} />
                <Button text="Créer" accessibilityLabel="Créer mon compte" onButtonPress={handleNextStep} />
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    title: {
        marginBottom: 20,
    },
    buttonsContainer: {
        marginTop: 30,
        flexDirection: 'row',
        justifyContent: 'space-between',
    },
});
