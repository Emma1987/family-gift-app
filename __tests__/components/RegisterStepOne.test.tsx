import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import { FormProvider, useForm } from 'react-hook-form';

import { RegisterStepOne } from '@/components/RegisterStepOne';

describe('RegisterStepOne', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    const Wrapper = ({ children }: { children: React.ReactNode }) => {
        const methods = useForm();
        return <FormProvider {...methods}>{children}</FormProvider>;
    };

    it('renders all input fields', () => {
        const { getByLabelText, getByText } = render(
            <Wrapper>
                <RegisterStepOne nextStep={jest.fn()} />
            </Wrapper>
        );

        expect(getByText('Étape 1 : Informations')).toBeTruthy();
        expect(getByLabelText('Nom')).toBeTruthy();
        expect(getByLabelText('Adresse email')).toBeTruthy();
        expect(getByLabelText('Mot de passe')).toBeTruthy();
        
        expect(getByText('Suivant')).toBeTruthy();
    });

    it('shows validation errors when submitting empty form', async () => {
        const { getByText } = render(
            <Wrapper>
                <RegisterStepOne nextStep={jest.fn()} />
            </Wrapper>
        );

        fireEvent.press(getByText('Suivant'));

        await waitFor(() => {
            expect(getByText('Le nom est obligatoire')).toBeTruthy();
            expect(getByText("L'email est obligatoire.")).toBeTruthy();
            expect(getByText('Le mot de passe est obligatoire.')).toBeTruthy();
        });
    });

    it('shows validation error when submitting a form with a name longer than 150 characters', async () => {
        const { getByText, getByLabelText } = render(
            <Wrapper>
                <RegisterStepOne nextStep={jest.fn()} />
            </Wrapper>
        );

        fireEvent.changeText(getByLabelText('Nom'), 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur sollicitudin felis sed tortor lacinia, vitae pharetra magna vestibulum. Etiam vel turpis a lorem pretium accumsan.');
        fireEvent.changeText(getByLabelText('Adresse email'), 'john@example.com');
        fireEvent.changeText(getByLabelText('Mot de passe'), 'Password1');

        fireEvent.press(getByText('Suivant'));

        await waitFor(() => {
            expect(getByText('Le nom ne doit pas dépasser 150 caractères')).toBeTruthy();
        });
    });

    it('shows validation error when submitting a form with an email that is not a valid email', async () => {
        const { getByText, getByLabelText } = render(
            <Wrapper>
                <RegisterStepOne nextStep={jest.fn()} />
            </Wrapper>
        );

        fireEvent.changeText(getByLabelText('Nom'), 'John Doe');
        fireEvent.changeText(getByLabelText('Adresse email'), 'john@example');
        fireEvent.changeText(getByLabelText('Mot de passe'), 'Password1');

        fireEvent.press(getByText('Suivant'));

        await waitFor(() => {
            expect(getByText("Cet email n'est pas un email valide")).toBeTruthy();
        });
    });

    it('shows validation error when submitting a form with a password that does not meet the requirements', async () => {
        const { getByText, getByLabelText } = render(
            <Wrapper>
                <RegisterStepOne nextStep={jest.fn()} />
            </Wrapper>
        );

        fireEvent.changeText(getByLabelText('Nom'), 'John Doe');
        fireEvent.changeText(getByLabelText('Adresse email'), 'john@example.com');
        fireEvent.changeText(getByLabelText('Mot de passe'), 'johndoe');

        fireEvent.press(getByText('Suivant'));

        await waitFor(() => {
            expect(getByText("Le mot de passe doit être composé d'au minimum 8 caractères, et comporter au moins 1 lettre majuscule, 1 lettre minuscule et 1 chiffre.")).toBeTruthy();
        });
    });

    it('calls nextStep when form is valid', async () => {
        const nextStepMock = jest.fn();

        const { getByText, getByLabelText } = render(
            <Wrapper>
                <RegisterStepOne nextStep={nextStepMock} />
            </Wrapper>
        );

        fireEvent.changeText(getByLabelText('Nom'), 'John Doe');
        fireEvent.changeText(getByLabelText('Adresse email'), 'john@example.com');
        fireEvent.changeText(getByLabelText('Mot de passe'), 'Password1');

        fireEvent.press(getByText('Suivant'));

        await waitFor(() => {
            expect(nextStepMock).toHaveBeenCalled();
        });
    });

    it('matches the snapshot', () => {
        const tree = render(
            <Wrapper>
                <RegisterStepOne nextStep={jest.fn()} />
            </Wrapper>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
