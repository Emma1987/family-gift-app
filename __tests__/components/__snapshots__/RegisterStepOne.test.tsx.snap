// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterStepOne matches the snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
      "marginTop": "25%",
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        undefined,
        undefined,
        undefined,
        {
          "fontSize": 20,
          "fontWeight": "bold",
        },
        undefined,
        {
          "marginBottom": 20,
        },
      ]
    }
  >
    Étape 1 : Informations
  </Text>
  <View>
    <TextInput
      accessibilityLabel="Nom"
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Nom"
      style={
        [
          {
            "borderBottomWidth": 1,
            "fontWeight": 500,
            "height": 45,
            "marginBottom": 20,
          },
          {
            "borderColor": "#151718",
            "color": "#151718",
          },
          undefined,
        ]
      }
    />
  </View>
  <View>
    <TextInput
      accessibilityLabel="Adresse email"
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="email-address"
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholder="Adresse email"
      style={
        [
          {
            "borderBottomWidth": 1,
            "fontWeight": 500,
            "height": 45,
            "marginBottom": 20,
          },
          {
            "borderColor": "#151718",
            "color": "#151718",
          },
          undefined,
        ]
      }
    />
  </View>
  <View>
    <View
      style={
        [
          {
            "alignItems": "center",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "marginBottom": 20,
          },
          {
            "borderColor": "#151718",
          },
          undefined,
        ]
      }
    >
      <TextInput
        accessibilityLabel="Mot de passe"
        autoCapitalize="none"
        autoComplete="password-new"
        autoCorrect={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        placeholder="Mot de passe"
        secureTextEntry={true}
        style={
          [
            {
              "flex": 1,
              "fontWeight": "500",
              "height": 45,
            },
            {
              "color": "#151718",
            },
          ]
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
            "padding": 10,
          }
        }
      >
        <Text
          accessibilityLabel="Changer la visibilité du mot de passe"
        >
          eye-off-outline
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "flex-end",
      }
    }
  >
    <View
      accessibilityLabel="Page suivante"
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          [
            {
              "backgroundColor": "#fffaf0",
            },
            [
              {
                "borderRadius": 8,
              },
              {
                "paddingHorizontal": 20,
                "paddingVertical": 10,
              },
              {
                "backgroundColor": "#151718",
              },
            ],
          ]
        }
        testID="button-inner-container"
      >
        <Text
          style={
            [
              {
                "color": "#11181C",
              },
              {
                "fontSize": 16,
                "lineHeight": 24,
              },
              undefined,
              undefined,
              undefined,
              undefined,
              [
                {
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                {
                  "color": "#f0e3c9",
                },
              ],
            ]
          }
        >
          Suivant
        </Text>
      </View>
    </View>
  </View>
</View>
`;
